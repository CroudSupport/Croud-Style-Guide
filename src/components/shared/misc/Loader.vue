<template>
    <div class="ui inverted dimmer" :class="{ active: loading }">
        <div class="ui text loader" :class="{ [size]: true }">
            <slot>
                Loading
            </slot>
        </div>
    </div>
</template>

<script>
   /**
    * A loading spinner combined with a dimmer
    *
    * @see https://semantic-ui.com/elements/loader.html
    * @example ./croud-loader.md
    */
    export default {
        name: 'croud-loader',

        props: {
           /**
            * Whether the loader is active or not
            */
            loading: {
                type: Boolean,
                required: true,
            },

           /**
            * Size of the loader
            * @see https://semantic-ui.com/elements/loader.html#size
            */
            size: {
                type: String,
                default: 'large',
            },
        },
    }
</script>
