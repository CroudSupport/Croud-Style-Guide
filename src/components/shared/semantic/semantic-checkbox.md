### semantic-checkbox
For our checkboxes, we use the semantic-checkbox component from the [croud-vue-semantic](https://github.com/CroudSupport/vue-semantic) library. Below are a few examples of semantic-checkbox in action. Click [here](http://croudsupport.github.io/vue-semantic/#semantic-checkbox) for full component docs.

### Basic usage

    <semantic-checkbox v-model="checkbox" label="🍕 Pizza" />

### Styling
Pass through the **type** prop to change the style of checkbo

    <div class="ui form">
        <div class="ui field">
            <semantic-checkbox v-model="checkbox" label="Checkbox" />
        </div>
        <div class="ui field">
            <semantic-checkbox v-model="checkbox" label="Slider" type="slider"/>
        </div>
        <div class="ui field">
            <semantic-checkbox v-model="checkbox" label="Toggle" type="toggle" />
        </div>
    </div>

### Disabled

    <semantic-checkbox label="🍕 Pizza" type="toggle" :disabled="true"/>
