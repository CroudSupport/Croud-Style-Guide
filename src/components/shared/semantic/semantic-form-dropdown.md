### semantic-form-dropdown
For our form dropdowns, we use the semantic-form-dropdown component from the [croud-vue-semantic](https://github.com/croudtech/vue-semantic) library. Below are a few examples of semantic-form-dropdown in action. Click [here](http://croudtech.github.io/vue-semantic/#semantic-form-dropdown) for full component docs.

### Basic usage

    <semantic-form-dropdown v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]"></semantic-form-dropdown>

### Styling
You can use the props to change the styling of the dropdown

    <div class="ui form">
        <div class="field">
            <label>Fluid</label>
            <semantic-form-dropdown fluid v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
        </div>

        <div class="three fields">
            <div class="field">
                <label>Search</label>
                <semantic-form-dropdown search v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
            </div>
            <div class="field">
                <label>Multiple</label>
                <semantic-form-dropdown multiple v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
            </div>
            <div class="field">
                <label>Disabled</label>
                <semantic-form-dropdown :disabled="true" v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
            </div>
        </div>

        <div class="inline field">
            <label>Inline</label>
            <semantic-form-dropdown :inline="true" v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
        </div>
    </div>

### Using slots
The **default** slot allows you to add custom options to the top of the dropdown

    <semantic-form-dropdown v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]">
        <div class="item" v-data="3">
            <b>Baz</b>
        </div>
    </semantic-form-dropdown>

### Handle Events
This component emits the **dropdown-selected** event when an item is selected.

    <semantic-form-dropdown :value="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" @dropdown-selected="$toasted.success('TaDa!!!')"></semantic-form-dropdown>
