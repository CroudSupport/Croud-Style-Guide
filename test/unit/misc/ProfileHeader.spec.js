import Vue from 'vue'
import Loader from '../../../src/components/shared/misc/ProfileHeader'
import '../../../semantic/dist/semantic'

Vue.use(require('croud-vue-semantic'))
Vue.component('CroudImageUploader', require('../../../src/components/shared/misc/ImageUploader'))
Vue.component('CroudAvatar', require('../../../src/components/shared/misc/Avatar'))

const Constructor = Vue.extend(Loader)

const vm = new Constructor({
    propsData: {
        user: {
            name: 'test',
            first_name: 'test',
            last_name: 'test',
            avatar: '',
            email: 'test',
        },
        readOnly: false,
    },
}).$mount()

describe('Profile header', () => {
    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    })

    describe('read only', () => {
        beforeEach(() => {
            vm.readOnly = true
        })

        it('should match the snapshot', () => {
            expect(vm.$el).toMatchSnapshot()
        })
    })
})
