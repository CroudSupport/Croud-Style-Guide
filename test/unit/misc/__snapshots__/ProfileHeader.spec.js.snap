// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile header read only should match the snapshot 1`] = `
<div
  class="ui stackable grid"
>
  <div
    class="three wide column"
  >
    <div
      requstheaders="[object Object]"
    >
      <div
        class="ui very padded center aligned container"
      >
        <div>
          <div
            class="ui circular image small"
          >
            <svg
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
              xml:space="preserve"
              viewBox="0 0 500 500"
              style="shape-rendering: geometricPrecision; text-rendering: geometricPrecision; image-rendering: optimizeQuality; fill-rule: evenodd;"
            >
              <g>
                <circle
                  cx="250"
                  cy="250"
                  r="245"
                  style="fill: #e4e7e9;"
                />
                <text
                  x="50%"
                  y="50%"
                  text-anchor="middle"
                  dy="0em"
                  font-size="250px"
                  alignment-baseline="central"
                  fill="white"
                >
                  T
                </text>
              </g>
            </svg>
          </div>
        </div>
        <!--  -->
      </div>
    </div>
  </div>
  <div
    class="thirteen wide column"
  >
    <div
      class="ui form user-details"
    >
      <div
        class="required field"
      >
        <div
          class="ui big transparent input"
        >
          <input
            type="text"
            placeholder="First Name"
            readonly="true"
          />
        </div>
      </div>
      <div
        class="required field"
      >
        <div
          class="ui big transparent input"
        >
          <input
            type="text"
            placeholder="Last Name"
            readonly="true"
          />
        </div>
      </div>
      <div
        class="required field"
      >
        <div
          class="ui big transparent input"
        >
          <input
            type="text"
            placeholder="Email"
            readonly="true"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Profile header should match the snapshot 1`] = `
<div
  class="ui stackable grid"
>
  <div
    class="three wide column"
  >
    <div
      requstheaders="[object Object]"
    >
      <div
        class="ui very padded center aligned container"
      >
        <div>
          <div
            class="ui circular image small"
          >
            <svg
              xmlns="http://www.w3.org/2000/svg"
              xmlns:xlink="http://www.w3.org/1999/xlink"
              xml:space="preserve"
              viewBox="0 0 500 500"
              style="shape-rendering: geometricPrecision; text-rendering: geometricPrecision; image-rendering: optimizeQuality; fill-rule: evenodd;"
            >
              <g>
                <circle
                  cx="250"
                  cy="250"
                  r="245"
                  style="fill: #e4e7e9;"
                />
                <text
                  x="50%"
                  y="50%"
                  text-anchor="middle"
                  dy="0em"
                  font-size="250px"
                  alignment-baseline="central"
                  fill="white"
                >
                  T
                </text>
              </g>
            </svg>
          </div>
        </div>
        <div>
          <div
            class="ui divider"
          />
          <button
            class="ui blue fluid button"
          >
            Set Image
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="thirteen wide column"
  >
    <div
      class="ui form user-details"
    >
      <div
        class="required field"
      >
        <div
          class="ui big transparent input"
        >
          <input
            type="text"
            placeholder="First Name"
          />
        </div>
      </div>
      <div
        class="required field"
      >
        <div
          class="ui big transparent input"
        >
          <input
            type="text"
            placeholder="Last Name"
          />
        </div>
      </div>
      <div
        class="required field"
      >
        <div
          class="ui big transparent input"
        >
          <input
            type="text"
            placeholder="Email"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
